components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -1161,7 +1161,10 @@ void EthereumProviderImpl::ContinueRequestEthereumPermissionsKeyringInfo(
     brave_wallet::mojom::KeyringInfoPtr keyring_info) {
   DCHECK_EQ(keyring_info->id, brave_wallet::mojom::kDefaultKeyringId);
   if (!keyring_info->is_keyring_created) {
-    delegate_->ShowWalletOnboarding();
+    if (!wallet_onboarding_shown_) {
+      delegate_->ShowWalletOnboarding();
+      wallet_onboarding_shown_ = true;
```

Comment:
do we need to add such protection for Solana Dapps?

Reply:
it's probably best ya but I'll wait for a site that reproduces for it so I can verify it is done right instead of building my own manual test case for QA